main.tf

```diff
@@ -217,6 +217,21 @@ resource "google_alloydb_instance" "read_pool" {
   labels            = var.primary_instance.labels
   annotations       = var.primary_instance.annotations
 
+  dynamic "network_config" {
+    for_each = each.value.enable_public_ip ? ["network_config"] : []
+    content {
+      enable_public_ip = each.value.enable_public_ip
+
+      dynamic "authorized_external_networks" {
+        for_each = each.value.cidr_range != null ? each.value.cidr_range : []
```

Comment:
@rayanebel I am wondering if we need to change `each.value.cidr_range` to `toset(each.value.cidr_range)` as `cidr_range` is defined as `list` and for_each accepts `set`.

Reply:
@imrannayer Done, I've also tested this part and it works